components/advanced_create_post/advanced_create_post.tsx

```diff
@@ -1029,67 +1029,162 @@ class AdvancedCreatePost extends React.PureComponent<Props, State> {
     }
 
     handleKeyDown = (e: React.KeyboardEvent<TextboxElement>) => {
```

Comment:
I wonder if we could somehow unify, or have a common util, for both create post and comment, to handle these events.
It could just return strings like 'quote', 'code', 'emoji' etc, and let the components handle those.
Does it make sense to you?

Reply:
the idea is to move all the formatting into the advanced text editor and just loop in a changeHandler for parent-component specific handling